autorest/preparer.go

```diff
@@ -183,6 +183,16 @@ func WithBaseURL(baseURL string) PrepareDecorator {
 	}
 }
 
+// WithCustomBaseURL returns a PrepareDecorator that replaces brace-enclosed keys within the
+// request base URL (i.e., http.Request.URL) with the corresponding values from the passed map.
+func WithCustomBaseURL(baseURL string, urlParameters map[string]interface{}) PrepareDecorator {
```

Comment:
Truth is it does not need to be interface. I'm following what `WithPathParameters` and `WithQueryParameters` do right now. Still will take a look if those can use `map[string]string`.